### PR TITLE
Fix empty table case for jaccard algo

### PR DESCRIPTION
**Pull request by YakhtinLeonid**
Changes proposed in this pull request:
Fix the jaccard algorithm for tables with zero columns and non-zero rows

### CODE REVIEW

File: cpp/oneapi/dal/algo/jaccard/detail/vertex_similarity_ops.hpp [hunk 1]
```diff
@@ -96,6 +96,13 @@ vertex_similarity_result vertex_similarity_ops_dispatcher<Policy, Float, Method,
     const std::int64_t max_block_size =
         get_max_block_size<Float, vertex_type<Graph>>(number_elements_in_block);
     void *result_ptr = input.get_caching_builder()(max_block_size);
+
+    if (!number_elements_in_block) {
+        homogen_table empty_table;
```
Comment (orrrrtem) [inline cpp/oneapi/dal/algo/jaccard/detail/vertex_similarity_ops.hpp:101]: Why do you need this?

File: cpp/oneapi/dal/algo/jaccard/detail/vertex_similarity_ops.hpp [hunk 2]
```diff
@@ -96,6 +96,13 @@ vertex_similarity_result vertex_similarity_ops_dispatcher<Policy, Float, Method,
     const std::int64_t max_block_size =
         get_max_block_size<Float, vertex_type<Graph>>(number_elements_in_block);
     void *result_ptr = input.get_caching_builder()(max_block_size);
+
+    if (!number_elements_in_block) {
```
Comment (orrrrtem) [inline cpp/oneapi/dal/algo/jaccard/detail/vertex_similarity_ops.hpp:100]: Bad if. Make it an explicit equation.

File: cpp/oneapi/dal/algo/jaccard/detail/vertex_similarity_ops.hpp [hunk 3]
```diff
@@ -96,6 +96,12 @@ vertex_similarity_result vertex_similarity_ops_dispatcher<Policy, Float, Method,
     const std::int64_t max_block_size =
         get_max_block_size<Float, vertex_type<Graph>>(number_elements_in_block);
     void *result_ptr = input.get_caching_builder()(max_block_size);
+
+    if (number_elements_in_block == 0) {
+        vertex_similarity_result res(homogen_table(), homogen_table(), 0);
```
Comment (michael-smirnov) [inline cpp/oneapi/dal/algo/jaccard/detail/vertex_similarity_ops.hpp:101]:
```suggestion
        vertex_similarity_result res(table(), table(), 0);
```
You can even stick just to the table, if you do not have a requirement that the result should always contain homogen tables.

Comment (michael-smirnov) [reply cpp/oneapi/dal/algo/jaccard/detail/vertex_similarity_ops.hpp:101]: But the better way I think is to have a default constructor for `vertex_similarity_result`

**Review by orrrrtem — COMMENTED** [review]
What is behavior on the user side? If user will request get_coeffs?

**Comment by YakhtinLeonid** [comment]
An empty table will be returned from get_coeffs. it can be printed without fails

**Comment by orrrrtem** [comment]
> An empty table will be returned from get_coeffs. it can be printed without fails

Ok.  Do you measure performance differentiation after the fix?

**Comment by orrrrtem** [comment]
Update Branch.

**Comment by YakhtinLeonid** [comment]
> > An empty table will be returned from get_coeffs. it can be printed without fails
> 
> Ok. Do you measure performance differentiation after the fix?

The performance is almost the same (there is a small speed up, no more than 0.5%)

**Review by orrrrtem — APPROVED** [review]

**Pull request merged**